src/EntityId/PrefixMappingEntityIdParser.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Wikibase\DataModel\Services\EntityId;
+
+use Wikimedia\Assert\Assert;
+use Wikimedia\Assert\ParameterAssertionException;
+use Wikimedia\Assert\ParameterTypeException;
+use Wikibase\DataModel\Entity\EntityId;
+use Wikibase\DataModel\Entity\EntityIdParser;
+
+/**
+ * EntityIdParser that adds a fixed prefix, maps a prefix of id serialization
+ * to a local prefix according to the prefix mapping
+ * and parses resulting string as an EntityId.
+ * This can be used to prefix IDs of entities coming from a foreign repository
+ * with the repository name to avoid clashes with IDs of local entities.
```

Comment:
(Not a blocker for this PR, but a TODO): This should reference a proper spec document. That spec should contain essentially what is described in <https://phabricator.wikimedia.org/T133381>. The spec should live in docs/foreign-entity-ids.wiki (or .md?), but I'm not sure in which repo. It could live in Wikibase, so we have all the docs in one place. But that would break with the idea that documentation should live in the same repo as code, and should always be kept in sync with the code, commit by commit. So we'' want a docs directory in the WikibaseDataModelServices I suppose.